src/v/archival/ntp_archiver_service.cc

```diff
@@ -716,8 +779,13 @@ ss::future<ntp_archiver::batch_result> ntp_archiver::wait_all_scheduled_uploads(
           total.num_succeded,
           total.num_failed);
 
-        auto res = co_await upload_manifest();
-        if (res != cloud_storage::upload_result::success) {
+        // There are two possible uploads of the manifest, one for compacted
+        // segments and another for non-compacted segments. It is possible to
+        // upload just once after both sets of uploads are finished, but the
+        // shard may lose leadership between function calls and the new leader
+        // will have to do the segment upload again.
```

Comment:
Actually, this is no longer the case.
After recent changes to `ntp_archiver` the new leader will use data from the its snapshot, so if metadata is replicated we won't be reuploading anything. So, it's totally OK to wait for all segment uploads to complete and then re-upload the manifest.

Reply:
changed to upload manifest once if any uploads are successful.